component/metrics/scrape/scrape.go

```diff
@@ -0,0 +1,209 @@
+package scrape
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/metrics"
+	"github.com/grafana/agent/pkg/build"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/scrape"
+)
+
+func init() {
+	scrape.UserAgent = fmt.Sprintf("GrafanaAgent/%s", build.Version)
+
+	component.Register(component.Registration{
+		Name: "metrics.scrape",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			return New(opts, args.(Arguments))
+		},
+	})
+	component.RegisterGoStruct("MetricsReceiver", metrics.Receiver{})
+}
+
+// Arguments holds values which are used to configure the metrics.scrape
+// component.
+type Arguments struct {
+	Targets   []Target            `hcl:"targets"`
+	Receivers []*metrics.Receiver `hcl:"receivers"`
+
+	// Scrape Options
+	ExtraMetrics bool `hcl:"extra_metrics,optional"`
+	// TODO(@tpaschalis) enable HTTPClientOptions []config_util.HTTPClientOption
+
+	// Scrape Config
+	ScrapeConfigs []Config `hcl:"scrape_config,block"`
```

Comment:
I don't think it makes a lot of sense for a scrape component to have more than one scrape job, especially if the set of targets are shared across all jobs. Do you have a use case in mind here?

Reply:
Hmm, mainly for multi-target exporters (and being able to present their grouped status), and only having a single instance of the scrape cache around. Come to think of it, I'm not sure the second point has any gains after all.

I see what you're saying though, and if it also enables us to avoid the issue with the shared Appenders then maybe it's the way to go.

Reply:
It doesn't avoid that problem :) There'd still be one scrape loop per target. I'm just not sure it makes sense to have many scrape jobs that share the same set of targets.